addons/hr_contract/models/hr_contract.py

```diff
@@ -247,7 +247,9 @@ def write(self, vals):
 
         calendar = vals.get('resource_calendar_id')
         if calendar:
-            self.filtered(lambda c: c.state == 'open' or (c.state == 'draft' and c.kanban_state == 'done')).mapped('employee_id').write({'resource_calendar_id': calendar})
+            self.filtered(
+                lambda c: c.state == 'open' or (c.state == 'draft' and c.kanban_state == 'done' and c.employee_id.contracts_count == 1)
+            ).mapped('employee_id').write({'resource_calendar_id': calendar})
```

Comment:
As we're fetching records here, I guess we can drop the mapped method :)
```suggestion
            ).employee_id.write({'resource_calendar_id': calendar})
```

Reply:
While we're at it, does it make sense to use this?

```suggestion
            ).employee_id.resource_calendar_id = calendar
```
It's using the update method rather than the write method. I personally think it looks cleaner.

Reply:
Indeed! Makes sense to me :)